types/json.go

```diff
@@ -26,7 +26,7 @@ func (b *JSONBuilder) AppendBytes(v []byte) {
 		return
 	}
 
-	b.ExtensionBuilder.Builder.(*array.BinaryBuilder).Append(v)
```

Comment:
I do think we need Binary here

Reply:
why?

Reply:
Although it is true, we do want to save the extra cycles of casting from/to json.RawMessage when we can.
Now it will cause the extra conversions and will result in unneeded allocations as well.